### DISCUSSION TITLE
Not able to run terraform plan and terraform apply from cli

### DISCUSSION THREAD

**Discussion by abhinavsquadstack** [terrakube-io/terrakube#515] [Q&A] [answered]
Hello Team, 
I am not able to run `terraform plan` and `terraform apply` from CLI. Although plan and apply works as intended when initialized from UI. 
Steps which i am doing from CLI
 - Run `terraform login <Web_API>` : This works fine and I am successfully getting redirected to authentication page and secret is stored in local credentials file
 - Run `terraform init`: This runs fine and if workspace with given name isn't in the terrakube it creates one
 - `terraform state push terraform.tfstate`: This successfully uploads my state file to terrakube with a new run(Dummy state upload complete)

After doing all above, while running terraform plan or terraform apply I am not able to connect
<img width="1470" alt="image" src="https://github.com/AzBuilder/terrakube/assets/121935932/e5f2d89a-fe87-4dcf-bbc2-7d0c2c3cab8c">

Please help. Am i missing something here?

Thanks

**Comment by alfespa17** [comment]
Hello @abhinavsquadstack what terrakube version are you using?

**Reply by abhinavsquadstack** [reply]
It is v2.15

**Reply by alfespa17** [reply]
Could you upgrade to 2.16? 
What storage backend are you using? 
Could you share the API logs when you get the error?

**Reply by alfespa17** [reply]
By the way for me it is working using 2.16.0.

I have this file:

```terraform
resource "null_resource" "previous" {}

resource "time_sleep" "wait_30_seconds" {
  depends_on = [null_resource.previous]

  create_duration = "30s"
}

# This resource will create (at least) 30 seconds after null_resource.previous
resource "null_resource" "next" {
  depends_on = [time_sleep.wait_30_seconds]
}
```

I run terraform apply  to create the state file in my local directory "terraform.tfstate"

After that I created a backup of the terraform.tfstate and create a new directory with this terraform file:

```terraform
terraform {
  cloud {
    organization = "aws"
    hostname = "8080-azbuilder-terrakube-fkxcnytz91y.ws-us105.gitpod.io"

    workspaces {
      tags = ["development", "metadata"]
    }
  }
}

resource "null_resource" "previous" {}

resource "time_sleep" "wait_30_seconds" {
  depends_on = [null_resource.previous]

  create_duration = "30s"
}

# This resource will create (at least) 30 seconds after null_resource.previous
resource "null_resource" "next" {
  depends_on = [time_sleep.wait_30_seconds]
}
```

I can migrate the the state from my directory

```bash
terraform login 8080-azbuilder-terrakube-fkxcnytz91y.ws-us105.gitpod.io
terraform init
terraform state push terraform.tfstate
```

And my plan is running as expected:

![image](https://github.com/AzBuilder/terrakube/assets/4461895/b9ff22b2-7e56-401c-9a59-fc675e2854ad)

**Reply by abhinavsquadstack** [reply, marked as answer]
While running `terraform plan`, following are the api logs
<img width="926" alt="image" src="https://github.com/AzBuilder/terrakube/assets/121935932/e6b3276e-baea-4250-b75b-9444f3c02957">
I saw that the `upload-url ...` line has https://https:://<API_URL>/remote, which is incorrect
I changed my hostname variable from env file and now it is running fine

Thanks for your help
